trio/_subprocess.py

```diff
@@ -218,17 +230,187 @@ def returncode(self):
+
+    * If the subprocess exits with a nonzero status code, indicating failure,
+      :func:`run_process` raises a :exc:`subprocess.CalledProcessError`
+      exception rather than returning normally. The captured outputs
+      are still available as the ``stdout`` and ``stderr`` attributes
+      of that exception.
+
+    To suppress the :exc:`~subprocess.CalledProcessError` on failure,
+    pass ``check=False``. To obtain different I/O behavior, use the
+    lower-level standard stream :ref:`redirection options <subprocess-options>`
+    ``stdin``, ``stdout``, and/or ``stderr``. (If you want one of the
+    subprocess's standard streams to go to the same place the parent
+    Trio process's corresponding stream goes, pass an explicit ``None``.)
+
+    It is an error to specify ``input`` if ``stdin`` is specified. If
+    ``stdout`` or ``stderr`` is specified (as something other than
+    ``subprocess.PIPE``), the corresponding attribute of the returned
+    :class:`~subprocess.CompletedProcess` object will be ``None``.
+
+    Args:
+      command (list or str): The command to run. Typically this is a
+          sequence of strings such as ``['ls', '-l', 'directory with spaces']``,
+          where the first element names the executable to invoke and the other
+          elements specify its arguments. With ``shell=True`` in the
+          ``**options``, or on Windows, ``command`` may alternatively
+          be a string, which will be parsed following platform-dependent
+          quoting rules.
+      input (bytes): The input to provide to the subprocess on its
+          standard input stream. If you want the subprocess's input
+          to come from something other than data specified at the time
+          of the :func:`run_process` call, you can specify a redirection
+          using the lower-level ``stdin`` option; then ``input`` must
+          be unspecified or None.
+      check (bool): If false, don't validate that the subprocess exits
+          successfully. You should be sure to check the
+          ``returncode`` attribute of the returned object if you pass
+          ``check=False``, so that errors don't pass silently.
+      task_status: This function can be used with ``nursery.start``.
+          If it is, it returns the :class:`Process` object, so that other tasks
+          can send signals to the subprocess or wait for it to exit.
+          They shouldn't try to send or receive on the subprocess's
+          input and output streams, because :func:`run_process` is
+          already doing that. Note that signals which terminate a
+          subprocess often result in a nonzero return code; you
+          probably want to pass ``check=False`` and do your own
+          more specific error check if you're planning on sending any.
+      **options: :func:`run_process` also accepts any :ref:`general subprocess
+          options <subprocess-options>` and passes them on to the
+          :class:`~trio.Process` constructor.
```

Comment:
I'm having trouble thinking of cases where getting access to the `Process` object is the natural way to accomplish something... if someone's using `run_process` to pause a task to run a process, then it feels like the natural way to cancel that is to, y'know, cancel it :-). And TOOWTDI?

There's a long tradition in other libraries having lots of kinds of "thing you run", and giving each their own ad hoc cancellation API, that I've tried to avoid succumbing to...

Reply:
I could imagine wanting to send SIGSTOP/SIGCONT, or to send some graceful shutdown signal (since we dropped the explicit support for that), or to have access to the PID for various reasons. I guess it's true that many of these wouldn't overlap with output capturing, and with no capturing and `check=False` run_process is basically just `async with Process(...) as proc: pass`, so maybe it's more of an attractive nuisance than it is a useful tool? Happy to remove the `start()` support if you think it's not worth it after reading this explanation.

Reply:
I think let's drop the `task_status` for now, and we can always add it back later when we have more experience.